### DISCUSSION TITLE
[Approved] Add a behavior (or attached property) to allow users the ability to close the soft keyboard when clicking outside of a view

### DISCUSSION THREAD

**Discussion by PureWeen** [CommunityToolkit/Maui#967] [New Feature Discussions]
There are a number of users that want to be able to tap into the nothingness and close the soft keyboard. 

https://github.com/dotnet/maui/issues/12002#issuecomment-1385826740

We didn't copy this behavior over from XF because the mechanism used to implement this behavior was to make the `PageRenderer` focusable/clickable which wreaked havoc on accessibility. 

I have a proposal here for some APIs we could add to MAUI
https://github.com/dotnet/maui/issues/12003
https://github.com/dotnet/maui/issues/11983

I feel like this type of behavior should live inside the community toolkit, so it can be refined and added to out of band from the MAUI. We can release these initial features inside a NET 7 version of the community toolkit. Since this is added on behavior it seems like it would survive and iterate better inside the toolkit vs inside the core code. 

The core MAUI code just needs to provide the APIs necessary to enable this type of feature. 

I already have a behavior built here
https://github.com/PureWeen/ShanedlerSamples/tree/main/ShanedlerSamples/Library/TapToCloseBehavior
and the iOS code can be easily removed from MAUI core and put into a behavior as well. 

```XAML
            <Entry Placeholder="Text Field" Text="Starting text" WidthRequest="300">
                <Entry.Behaviors>
                    <local:SoftKeyboardBehavior CloseOnTapOutsideBounds="true"></local:SoftKeyboardBehavior>
                </Entry.Behaviors>
            </Entry>
```

**Comment by rachelkang** [comment]
I agree! Let's introduce these behaviors in toolkit to enable our developer community while preserving and continually encouraging accessibility

**Comment by pictos** [comment]
@PureWeen, so far this issue only happens on android?

**Reply by PureWeen** [reply]
It's only happening on Android because we left that code in place on iOS
https://github.com/dotnet/maui/pull/10948

So, we'd 
- Add this behavior in MCT that users could use for NET7 to make Android and iOS consistent
- NET8 I'd just remove this behavior from MAUI.Core so that Maui.Core is consistent.

**Reply by pictos** [reply]
Great, the API spec will be the same as on https://github.com/dotnet/maui/issues/11983?

**Reply by PureWeen** [reply]
I think eventually yes. 
That would be some additional features that could be added. 

The initial spec could include a couple of those as well but my thinking for the first version of this could just be something like

```XAML
            <Entry Placeholder="Text Field" Text="Starting text" WidthRequest="300">
                <Entry.Behaviors>
                    <local:SoftKeyboardBehavior CloseOnTapOutsideBounds="true"></local:SoftKeyboardBehavior>
                </Entry.Behaviors>
            </Entry>
```

And then it could grow to include the APIs talked about [here](https://github.com/dotnet/maui/issues/11983) as well.

**Reply by PureWeen** [reply]
So, the first feature would be this
https://github.com/dotnet/maui/issues/12003

Once that's merged and created then we could look at adding additional features like this
https://github.com/dotnet/maui/issues/11983

**Reply by pictos** [reply]
@PureWeen how can we implement the `interface` on our side and make that work on `Entry`? Should we create a custom Entry that implements that `interface` or that can be done with `AttachedProperty`?

**Reply by PureWeen** [reply]
The interface isn't really needed.

We'd only do an interface if this was made part of MAUi.Core

**Comment by pictos** [comment]
I approve this feature ✅

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
